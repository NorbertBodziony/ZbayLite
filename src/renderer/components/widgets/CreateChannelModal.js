import React from 'react'
import * as R from 'ramda'

import IconButton from '@material-ui/core/IconButton'
import { withStyles } from '@material-ui/core/styles'

import AddIcon from '@material-ui/icons/Add'

const styles = {
  icon: {
    fontSize: '26px'
  },
  button: {
    padding: '4px'
  }
}

export const CreateChannelModal = ({ classes, openModal }) => (
  <IconButton color='inherit' onClick={openModal} className={classes.button}>
    <AddIcon fontSize='inherit' className={classes.icon} />
  </IconButton>
)

export default R.compose(
  withStyles(styles),
  React.memo
)(CreateChannelModal)
