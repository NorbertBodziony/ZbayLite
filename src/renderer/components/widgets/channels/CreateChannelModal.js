import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'

import Grid from '@material-ui/core/Grid'
import { withStyles } from '@material-ui/core/styles'

import Modal from '../../ui/Modal'
import CreateChannelForm from '../../../containers/widgets/channels/CreateChannelForm.js'

const styles = theme => ({
  root: {
    padding: `${8 * theme.spacing.unit}px ${16 * theme.spacing.unit}px`,
    height: '100%',
    width: '100%'
  }
})

export const CreateChannelModal = ({ classes, open, handleClose }) => (
  <Modal
    open={open}
    handleClose={handleClose}
    title='New Channel'
    fullPage
  >
    <Grid className={classes.root}>
      <CreateChannelForm />
    </Grid>
  </Modal>
)

CreateChannelModal.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(CreateChannelModal)
