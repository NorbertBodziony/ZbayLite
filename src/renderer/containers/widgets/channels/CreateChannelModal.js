import { withModal } from '../../../store/handlers/modals'
import CreateChannelModal from '../../../components/widgets/channels/CreateChannelModal'

export default withModal('createChannel')(CreateChannelModal)
