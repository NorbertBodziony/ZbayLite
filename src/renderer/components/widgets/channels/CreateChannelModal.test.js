import React from 'react'
import { shallow } from 'enzyme'

import { CreateChannelModal } from './CreateChannelModal'
import { mockClasses } from '../../../../shared/testing/mocks'

describe('CreateChannelModal', () => {
  it('renders component', () => {
    const result = shallow(
      <CreateChannelModal
        handleClose={jest.fn()}
        open
        classes={mockClasses}
      />
    )
    expect(result).toMatchSnapshot()
  })

  it('renders closed component', () => {
    const result = shallow(
      <CreateChannelModal
        handleClose={jest.fn()}
        open={false}
        classes={mockClasses}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
