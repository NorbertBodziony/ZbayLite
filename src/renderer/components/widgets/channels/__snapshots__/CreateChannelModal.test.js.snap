// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateChannelModal renders closed component 1`] = `
<Memo(WithStyles(Modal))
  fullPage={true}
  handleClose={[MockFunction]}
  open={false}
  title="New Channel"
>
  <WithStyles(ForwardRef(Grid))
    className="root-mock"
  >
    <Connect(Component) />
  </WithStyles(ForwardRef(Grid))>
</Memo(WithStyles(Modal))>
`;

exports[`CreateChannelModal renders component 1`] = `
<Memo(WithStyles(Modal))
  fullPage={true}
  handleClose={[MockFunction]}
  open={true}
  title="New Channel"
>
  <WithStyles(ForwardRef(Grid))
    className="root-mock"
  >
    <Connect(Component) />
  </WithStyles(ForwardRef(Grid))>
</Memo(WithStyles(Modal))>
`;
